Bug: Upgrade to 8.5.0 breaks linting setup with NX and `projectService: true`

### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Issue Description

After upgrading `typescript-eslint` to the latest version `8.5.0` my linting setup with NX broke. 

The respective error message is `Parsing error: Could not read project service default project 'tsconfig.json': error TS5012: Cannot read file 'tsconfig.json': ENOENT: no such file or directory`. The respective files are included in a `tsconfig.lib.json` file which is being referenced from `tsconfig.json`. The same applies to files included by `tsconfig.spec.ts` (also referenced by `tsconfig.json`).

I am using `projectService: true` with a NX monorepo (see reproduction repo). The same setup worked fine with version `8.4.0`. The breaking change seems to be #9893.

### Reproduction Repository Link

https://github.com/hoi4/typescript-eslint-issue

### Repro Steps

1) clone the repo
2) `npm i`
3) `npx nx lint nx19 --skip-nx-cache`

Alternatively, when using `eslint` directly with an example file: `npx eslint apps/nx19/src/app/app.component.ts`

The latest commit uses version `8.5.0` which has the described issue. The previous commit holds a working state with version `8.4.0`. 


### Versions

<!-- Fill in the relevant versions and delete the others -->
| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `8.5.0` |
| `@typescript-eslint/parser`            | `8.5.0` |
| `@typescript-eslint/utils`             | `8.5.0` |
| `TypeScript`                           | `~5.5.2` |
| `ESLint`                               | `9.10.0` |
| `node`                                 | `20.15.0` |

> `Could not read project service default project 'tsconfig.json': error TS5012: Cannot read file 'tsconfig.json': ENOENT: no such file or directory`

Aha! This is from #9893. That PR landed under the assumption if that the cwd being linted _should_ generally have a `tsconfig.json` in that directory. Which https://github.com/hoi4/typescript-eslint-issue does not. It just has a `tsconfig.base.json` that other TSConfigs fall under.

Which brings up a question: _why_ is there not a root-level `tsconfig.json`? Are you just relying on the default editor type info for root-level files such as `eslint.config.js` and `jest.config.ts`?

--- a/packages/typescript-estree/src/create-program/createProjectService.ts
+++ b/packages/typescript-estree/src/create-program/createProjectService.ts
@@ -42,9 +42,10 @@ export function createProjectService(
   jsDocParsingMode: ts.JSDocParsingMode | undefined,
   tsconfigRootDir: string | undefined,
 ): ProjectServiceSettings {
+  const optionsRawObject = typeof optionsRaw === 'object' ? optionsRaw : {};
   const options = {
     defaultProject: 'tsconfig.json',
-    ...(typeof optionsRaw === 'object' && optionsRaw),
+    ...optionsRawObject,
   };
   validateDefaultProjectForFilesGlob(options.allowDefaultProject);
 
@@ -126,7 +127,7 @@ export function createProjectService(
   });
 
   log('Enabling default project: %s', options.defaultProject);
-  let configFile: ts.ParsedCommandLine;
+  let configFile: ts.ParsedCommandLine | undefined;
 
   try {
     configFile = getParsedConfigFile(
@@ -135,18 +136,22 @@ export function createProjectService(
       tsconfigRootDir,
     );
   } catch (error) {
-    throw new Error(
-      `Could not read project service default project '${options.defaultProject}': ${(error as Error).message}`,
-    );
+    if (optionsRawObject.defaultProject) {
+      throw new Error(
+        `Could not read project service default project '${options.defaultProject}': ${(error as Error).message}`,
+      );
+    }
   }
 
-  service.setCompilerOptionsForInferredProjects(
-    // NOTE: The inferred projects API is not intended for source files when a tsconfig
-    // exists.  There is no API that generates an InferredProjectCompilerOptions suggesting
-    // it is meant for hard coded options passed in. Hard asserting as a work around.
-    // See https://github.com/microsoft/TypeScript/blob/27bcd4cb5a98bce46c9cdd749752703ead021a4b/src/server/protocol.ts#L1904
-    configFile.options as ts.server.protocol.InferredProjectCompilerOptions,
-  );
+  if (configFile) {
+    service.setCompilerOptionsForInferredProjects(
+      // NOTE: The inferred projects API is not intended for source files when a tsconfig
+      // exists.  There is no API that generates an InferredProjectCompilerOptions suggesting
+      // it is meant for hard coded options passed in. Hard asserting as a work around.
+      // See https://github.com/microsoft/TypeScript/blob/27bcd4cb5a98bce46c9cdd749752703ead021a4b/src/server/protocol.ts#L1904
+      configFile.options as ts.server.protocol.InferredProjectCompilerOptions,
+    );
+  }
 
   return {
     allowDefaultProject: options.allowDefaultProject,

--- a/packages/typescript-estree/tests/lib/createProjectService.test.ts
+++ b/packages/typescript-estree/tests/lib/createProjectService.test.ts
@@ -65,7 +65,7 @@ describe('createProjectService', () => {
     expect(settings.allowDefaultProject).toBeUndefined();
   });
 
-  it('throws an error with a local path when options.defaultProject is not provided and getParsedConfigFile throws a diagnostic error', () => {
+  it('does not throw an error when options.defaultProject is not provided and getParsedConfigFile throws a diagnostic error', () => {
     mockGetParsedConfigFile.mockImplementation(() => {
       throw new Error('tsconfig.json(1,1): error TS1234: Oh no!');
     });
@@ -78,9 +78,7 @@ describe('createProjectService', () => {
         undefined,
         undefined,
       ),
-    ).toThrow(
-      /Could not read project service default project 'tsconfig.json': .+ error TS1234: Oh no!/,
-    );
+    ).not.toThrow();
   });
 
   it('throws an error with a relative path when options.defaultProject is set to a relative path and getParsedConfigFile throws a diagnostic error', () => {
@@ -132,6 +130,7 @@ describe('createProjectService', () => {
       createProjectService(
         {
           allowDefaultProject: ['file.js'],
+          defaultProject: 'tsconfig.json',
         },
         undefined,
         undefined,

